lib/blpapi-wrapper.ts

```diff
@@ -445,13 +421,29 @@ export class Session extends events.EventEmitter {
                 if ('options' in s) {
                     result.options = s.options;
                 }
+
                 return result;
-            }
+            });
 
-            var subs = _.map(subscriptionsAndServices, makeSubscription, this);
             // TODO: blpapi-node doesn't accept null; remove this when that's fixed.
             identity = identity || undefined;
             this.session.subscribe(subs, identity);
+
+            _.forEach(subs, (s: blpapi.Subscription): void => {
+                var uri = securityToService(s.security);
+                var cid = s.correlation;
+                var userSubscription = this.subscriptions[cid];
+
+                assert(uri in SERVICE_TO_SUBSCRIPTION_EVENTS_MAP,
+                       util.format('Service, %s, not handled', uri));
+                var events = SERVICE_TO_SUBSCRIPTION_EVENTS_MAP[uri];
+                events.forEach((event: string): void => {
+                    log('listening on event: ' + event + ' for cid: ' + cid);
+                    this.listen(event, cid, (m: any): void => {
+                        userSubscription.emit('data', m.data, s);
+                    });
+                });
+            });
         }).nodeify(cb);
```

Comment:
We'll need a `catch` block here to clean up cached `this.subscription[cid]` when `session.subscribe` throw.